server/src/main/java/org/elasticsearch/cluster/block/ClusterBlocks.java

```diff
@@ -68,27 +67,37 @@ public Map<String, Set<ClusterBlock>> indices(ClusterBlockLevel level) {
     }
 
     private Set<ClusterBlock> blocksForIndex(ClusterBlockLevel level, String index) {
-        return indices(level).getOrDefault(index, emptySet());
+        return indices(level).getOrDefault(index, Set.of());
     }
 
     private static EnumMap<ClusterBlockLevel, ImmutableLevelHolder> generateLevelHolders(
         Set<ClusterBlock> global,
         Map<String, Set<ClusterBlock>> indicesBlocks
     ) {
         EnumMap<ClusterBlockLevel, ImmutableLevelHolder> levelHolders = new EnumMap<>(ClusterBlockLevel.class);
+        List<ClusterBlock> setBuilder = new ArrayList<>();
```

Comment:
It seems that `setBuilder` list works like local accumulator for `addBlocksAtLevel` method, why not move it there?

Reply:
I think it's to avoid allocating a new `ArrayList` on each iteration - this way we allocate once and just keep re-using the same object for everything (noting that `ArrayList#clear` is _much_ cheaper).

But it's a good question, this deserves a source comment to help future readers.